backend/src/mirrors_qa_backend/routes/tests.py

```diff
@@ -1,57 +1,108 @@
-                "totalRecords": 20,
-            },
-        },
-        status_code=status.HTTP_200_OK,
+def list_tests(
+    session: DbSession,
+    worker_id: Annotated[str | None, Query()] = None,
+    country: Annotated[str | None, Query(min_length=3)] = None,
+    status: Annotated[list[StatusEnum] | None, Query()] = None,
+    page_size: Annotated[
+        int, Query(le=Settings.MAX_PAGE_SIZE, ge=1)
+    ] = Settings.MAX_PAGE_SIZE,
+    page_num: Annotated[int, Query(ge=1)] = 1,
+    sort_by: Annotated[TestSortColumnEnum, Query()] = TestSortColumnEnum.requested_on,
+    order: Annotated[SortDirectionEnum, Query()] = SortDirectionEnum.asc,
+) -> schemas.TestsList:
+    result = db.tests.list_tests(
+        session,
+        worker_id=worker_id,
+        country=country,
+        status=status,
+        page_size=page_size,
+        page_num=page_num,
+        sort_column=sort_by,
+        sort_direction=order,
+    )
+    return schemas.TestsList(
+        tests=[serializer.serialize_test(test) for test in result.tests],
+        metadata=schemas.calculate_pagination_metadata(
+            result.nb_tests, page_size=page_size, current_page=page_num
+        ),
     )
 
 
 @router.get(
     "/{test_id}",
-    status_code=status.HTTP_200_OK,
+    status_code=status_codes.HTTP_200_OK,
     responses={
-        status.HTTP_200_OK: {"description": "Returns the details of a test."},
-        status.HTTP_404_NOT_FOUND: {"description": "Test with id does not exist."},
+        status_codes.HTTP_200_OK: {"description": "Returns the details of a test."},
+        status_codes.HTTP_404_NOT_FOUND: {
+            "description": "Test with id does not exist."
+        },
     },
 )
-def get_test(test_id: str) -> Response:
-    return JSONResponse(
-        content={"id": test_id},
-        status_code=status.HTTP_200_OK,
-    )
+def get_test(test_id: UUID4, session: DbSession) -> schemas.Test:
+    test = db.tests.get_test(session, test_id)
+    if test is None:
+        raise http_errors.NotFoundError(f"Test with id '{test_id}' does not exist.")
+    return serializer.serialize_test(test)
 
 
 @router.patch(
     "/{test_id}",
-    status_code=status.HTTP_200_OK,
+    status_code=status_codes.HTTP_200_OK,
     responses={
-        status.HTTP_200_OK: {"description": "Update the details of a test."},
+        status_codes.HTTP_200_OK: {"description": "Update the details of a test."},
     },
 )
-def update_test(test_id: str) -> Response:
-    return JSONResponse(
-        content={"id": test_id},
-        status_code=status.HTTP_200_OK,
+def update_test(
+    session: DbSession,
+    worker: CurrentWorker,
+    test_id: UUID4,
+    update: schemas.UpdateTestModel,
+) -> schemas.Test:
+    data = update.model_dump(exclude_unset=True)
+    body = schemas.UpdateTestModel().model_copy(update=data)
+    # Ensure that the worker is the one who the test belongs to
+    test = db.tests.get_test(session, test_id)
+    if test is None:
```

Comment:
What about a Depends that returns a Test from test_id, raising 404 if missing or 401 if incorrect worker?